core/src/connection/manager/task.rs

```diff
@@ -0,0 +1,333 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+use crate::{
+    muxing::StreamMuxer,
+    connection::{Close, Connected, ConnectionError},
+};
+use futures::{prelude::*, channel::mpsc, stream};
+use std::{pin::Pin, task::Context, task::Poll};
+use super::{Connection, ConnectionHandler, ConnectResult, IntoConnectionHandler, Substream};
+
+/// Identifier of a [`Task`] in a [`Manager`](super::Manager).
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub struct TaskId(pub(super) usize);
+
+/// Commands that can be sent to a [`Task`].
+#[derive(Debug)]
+pub enum Command<T> {
+    /// Notify the connection handler of an event.
+    NotifyHandler(T),
+}
+
+/// Events that a task can emit to its manager.
+#[derive(Debug)]
+pub enum Event<T, H, TE, HE, C> {
```

Comment:
```suggestion
pub enum TaskEvent<T, H, TE, HE, C> {
```

As per my other comment.

Reply:
While my personal preference goes in the other direction, i.e. not to repeat enclosing module names within type names in the module, I'm also  not really keen on debating it, I really just tried to stay somewhat consistent with the current naming scheme and not change names where I don't see a strong enough incentive. In this case, it was actually just called `Event` before, i.e. I didn't change it. Given that this is also not in the public API, would you still like me to change it in this PR?

Reply:
Since it's a private API it doesn't matter so much, so let's skip this.